### DISCUSSION TITLE
Is there a way to get the source key representation?

### DISCUSSION THREAD

**Discussion by ip1981** [toml-rs/toml#745] [Q&A] [answered]
Let's have a TOML file
```toml
[foo]
"x\ny" = 42
'*' = 33
"?" = 121
```

What I want is that, after the file is parsed, I can get the table's keys in the original form as they are written in the file: `"x\ny"`, `'*'`, `"?"`. I mean literally `"x\ny"` , with backslash+n. Quotes maybe omitted, but I want to know what they were: single, double, none, etc.

P. S. Including in table headers:
```toml
[foo."*".bar]

[foo.'*'.bar]
```

**Comment by epage** [comment, marked as answer]
You need access to a `Key`, for example
- [`Table::key`](https://docs.rs/toml_edit/latest/toml_edit/struct.Table.html#method.key)
- [`Table::get_key_value`](https://docs.rs/toml_edit/latest/toml_edit/struct.Table.html#method.get_key_value)
- [`Table::get_values`](https://docs.rs/toml_edit/latest/toml_edit/struct.Table.html#method.get_values)

**Reply by ip1981** [reply]
Nice )
